Spigot-API-Patches/0269-Add-dropLeash-variable-to-EntityUnleashEvent.patch

```diff
@@ -0,0 +1,78 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: KennyTV <jahnke.nassim@gmail.com>
+Date: Fri, 29 Jan 2021 15:13:04 +0100
+Subject: [PATCH] Add dropLeash variable to EntityUnleashEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/entity/EntityUnleashEvent.java b/src/main/java/org/bukkit/event/entity/EntityUnleashEvent.java
+index a33986a0c437a673435206fc337031a7eebdab3b..bcf773ec65d9fc27bb8e433c2f676ab0d60c35ee 100644
+--- a/src/main/java/org/bukkit/event/entity/EntityUnleashEvent.java
++++ b/src/main/java/org/bukkit/event/entity/EntityUnleashEvent.java
+@@ -10,10 +10,19 @@ import org.jetbrains.annotations.NotNull;
+ public class EntityUnleashEvent extends EntityEvent {
+     private static final HandlerList handlers = new HandlerList();
+     private final UnleashReason reason;
++    private boolean dropLeash; // Paper
+ 
++    // Paper start - drop leash variable
++    @Deprecated
+     public EntityUnleashEvent(@NotNull Entity entity, @NotNull UnleashReason reason) {
++        this(entity, reason, false);
```

Comment:
Do we not want the leash to be dropped by default?

Reply:
If someone manually calls that event, I'm gonna assume that boolean will be irrevelant/"undefined" anyways

Reply:
Pretty sure we'd want it to drop by default, I can't really think of a situation where that shouldn't be the case.

Reply:
That constructor is not used in the server, so it has nothing to do with "dropping it by default"... because it will be unused. And besides, the server does have cases where it isn't dropped as well

Reply:
Looking at the NMS part, true's used 4 or 5 times where there's one or two situations where it could be true OR false.
It would most likely be better to just make it true by default and then use this constructor in NMS and in general while using the other for if it could be one or the other